core/src/test/java/hudson/model/FingerprintCleanupThreadTest.java

```diff
@@ -90,6 +97,32 @@ public void testIOExceptionOnLoad() throws IOException {
         assertTrue("Should have logged IOException.", logOutput.contains("ERROR: Failed to process"));
     }
 
+    @Test
+    public void testBlockingFacetBlocksDeletion() throws IOException {
+        createFolderStructure();
+        TestTaskListener testTaskListener = new TestTaskListener();
+        Fingerprint fp = new TestFingerprint(false);
+        fp.facets.setOwner(Saveable.NOOP);
+        TestFingperprintFacet facet = new TestFingperprintFacet(fp, System.currentTimeMillis(), true);
+        fp.facets.add(facet);
+        FingerprintCleanupThread cleanupThread = new TestFingerprintCleanupThread(fp);
+        cleanupThread.execute(testTaskListener);
+        assertThat(fpFile.toFile(), is(aReadableFile()));
```

Comment:
Would it make sense to assert log output here, given what this change does (at least "blocked deletion")?